Translate documentation to :gb: (#9)
europace/kex-push: README.md

> # Push-Mechanism for Changes to Vorgänge in KreditSmart

Reviewer: ```suggestion
# Push-Mechanism for changes to Vorgänge in KreditSmart
```

> > ⚠️ You'll find German domain-terms in the documentation, for translations and further explanations please refer to our [glossary](https://docs.api.europace.de/common/glossary/)

Reviewer 2: ```suggestion
> ⚠️ You'll find German domain-specific terms in the documentation, for translations and further explanations please refer to our [glossary](https://docs.api.europace.de/common/glossary/)
```

Author: wir haben diesen Satz generell besprochen und ich würde den gerne einheitlich halten.....

Reviewer 2: Ist `domain-term` angemessenes Englisch?

> An external client (public subscriber) registers at EUROPACE und receives certificates to be able establish a connection to AWS. For Echtgeschäft and Testumgebung you will receive separate certificates.

Reviewer 2: ```suggestion
An external client (public subscriber) registers at EUROPACE und receives certificates to establish a connection to the broker in AWS. For Echtgeschäft and Testumgebung you will receive separate certificates.
```

Author: borker oder broker? 🤔  Ich mag das abkürzen nicht, das kann man in privaten emails machen, aber bitte nicht in doku 😅

Reviewer 2: Den `borker` habe ich korrigiert 😓 

Die Contraction ist auch entfernt, wobei ich denke, dass man sie auch in öffentlichen Texten nutzen kann, um eben keine allzu hohe "Distanz" zu den Lesern aufzubauen. Die ausführliche Form ist jedoch immer die _richtige_, von daher 👍

> You can get the certificates from your contact in the KreditSmart team.

Reviewer 2: ```suggestion
You can get the certificates from your contact at the KreditSmart team.
```

Author: at? 

Vielleicht dann eher das hier?

```suggestion
You can get the certificates from your KreditSmart contact.
```

Reviewer 2: Ja, gefällt mir besser - weniger ist mehr 👍

[AWS](https://docs.aws.amazon.com/iot/latest/developerguide/aws-iot-how-it-works.html) notifies the client based on:

- [Topic Subscription](https://docs.aws.amazon.com/iot/latest/developerguide/topics.html)
- configuration changes ([Device Shadow](https://docs.aws.amazon.com/iot/latest/developerguide/iot-device-shadows.html))

> ## The partner tree as Topic

Reviewer: ```suggestion
## The partner tree as topic
```

> A Subscriber is allowed to receive all topics for Plaketten in the Partner tree underneath their own Plakette and for themselves. In addition the client is allowed to request a shadow document. This document povides you with the highest possible Topic. Should a topic change - e.g. due to restruring inside the partner tree - the client has to react appropriately. It is recommended to then `unsubsribe` to the old topic and `subscribe` to the new topic..

Reviewer 2: ```suggestion
A Subscriber is allowed to receive all topics for Plaketten in the Partner tree underneath their own Plakette and for themselves. In addition the client is allowed to request a shadow document. This document povides you with the least specific Topic. Should a topic change - e.g. due to restructuring inside the partner tree - the client has to react appropriately. It is recommended to then `unsubscribe` from the old topic and `subscribe` to the new topic.
```

Reviewer: Partner tree vs partner tree, sollte immer gleich geschrieben werden.

Author: @sgrell guter Punkt. was ist besser? partner tree oder?

Reviewer: "partner tree" wäre dann die englische Variante, fände ich hier passend.

> A Subscriber is allowed to receive all topics for Plaketten in the partner tree underneath their own Plakette and for themselves. In addition the client is allowed to request a shadow document. This document povides you with the highest possible Topic. Should a topic change - e.g. due to restruring inside the partner tree - the client has to react appropriately. It is recommended to then `unsubsribe` to the old topic and `subscribe` to the new topic..

Reviewer: ```suggestion
A Subscriber is allowed to receive all topics for Plaketten in the partner tree underneath their own Plakette and for themselves. In addition the client is allowed to request a shadow document. This document povides you with the highest possible topic. Should a topic change - e.g. due to restructuring inside the partner tree - the client has to react appropriately. It is recommended to then `unsubscribe` to the old topic and `subscribe` to the new topic.
```

For the topics the [AWS Limits](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html#iot-protocol-limits) apply, for example the depth can be a maximum of 7 layers.

> To receive changes not only for the own Plakette but for all other Plaketten _underneath_ the own Plakette you need to keep in mind to use a hash `#` as a wildcard. For example is your certificate valid for the topic `ECHTGESCHAEFT/PARTNER1/PARTNER2/PARTNER3` the subscriber needs to be configured for the topic `ECHTGESCHAEFT/PARTNER1/PARTNER2/PARTNER3/#`. You can find more details in the developer guide about [Topic Subscription](https://docs.aws.amazon.com/iot/latest/developerguide/topics.html).

Reviewer: the own klingt irgendwie sehr eigenartig 
```suggestion
To receive changes not only for your own Plakette but for all other Plaketten _underneath_ your own Plakette you need to use a hash `#` as a wildcard. For example, if your certificate is valid for the topic `ECHTGESCHAEFT/PARTNER1/PARTNER2/PARTNER3` the subscriber needs to be configured for the topic `ECHTGESCHAEFT/PARTNER1/PARTNER2/PARTNER3/#`. You can find more details in the developer guide about [Topic Subscription](https://docs.aws.amazon.com/iot/latest/developerguide/topics.html).
```

> To receive changes not only for the own Plakette but for all other Plaketten _underneath_ the own Plakette you need to keep in mind to use a hash `#` as a placeholder. For example is your ceteficate valid for the topic `ECHTGESCHAEFT/PARTNER1/PARTNER2/PARTNER3` the subscriber needs to be configured for the topic `ECHTGESCHAEFT/PARTNER1/PARTNER2/PARTNER3/#`. More details about this you can find in the documentation to [Topic Subscription](https://docs.aws.amazon.com/iot/latest/developerguide/topics.html).

Reviewer 2: ```suggestion
To receive changes not only for the own Plakette but for all other Plaketten _underneath_ the own Plakette you need to keep in mind to use a hash `#` as a wildcard. For example is your certificate valid for the topic `ECHTGESCHAEFT/PARTNER1/PARTNER2/PARTNER3` the subscriber needs to be configured for the topic `ECHTGESCHAEFT/PARTNER1/PARTNER2/PARTNER3/#`. You can find more details in the developer guide about [Topic Subscription](https://docs.aws.amazon.com/iot/latest/developerguide/topics.html).
```

## Payload Message Format

> The message body contains a payload in JSON format:

Reviewer 2: ```suggestion
The message body contains a JSON encoded payload:
```

Author: encoded? Das klingt komisch, dann eher a JSON formatted payload?

Reviewer 2: Wir könnten auch auf `content type` ausweichen ;)

Author: ```suggestion
The message body contains the payload as JSON:
```

Reviewer 2: 👍

}
```

> More details to the attributes:

Reviewer 2: ```suggestion
More details about the attributes:
```

- `vorgangsnummer`:

>   For changes in either Vorgang or Antrag the KEX-PUSH message always only contains the Vorgangsnummer.

Reviewer 2: ```suggestion
  For changes of both Vorgang or Antrag the KEX-PUSH message contains only the Vorgangsnummer.
```

Author: Vielleicht so?

```suggestion
  For changes of both Vorgang and Antrag the KEX-PUSH message only contains the Vorgangsnummer.
```

Reviewer 2: jetzt ist das `and` irreführend: das klingt so, als ginge es darum, das Vorgang und Antrag _gleichzeitig_ geändert werden.

Author: jaja nicht nur deutsch ist schwierig 😅 

Ich würde es dann gerne so lassen :)

Reviewer: ```suggestion
  For changes in either Vorgang or Antrag the KEX-Push message always only contains the Vorgangsnummer.
```

- `kundenbetreuerPartnerbaum`:

>   Because of the [limits for Topics](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html#iot-protocol-limits) to a maximium of 7 it can happen that a topic gets cut off if the corresponding Partner tree is too deep. To still be able to notify the subscriber about a Kundenbetreuer, the property `kundenbetreuerPartnerbaum` contains the full tree.

Reviewer 2: ```suggestion
  Because of the [limits for Topics](https://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html#iot-protocol-limits) to a maximium of 7 it can happen that a topic gets cut off if the corresponding Partner tree is too deep. To still be able to notify the subscriber about a Kundenbetreuer, the property `kundenbetreuerPartnerbaum` contains the complete tree.
```

- `letztesAenderungsDatum`:

  This property is always the current timestamp at the time the message was sent.

- `quellsystem`:

>   Currently only `KREDITSMART` is supported as a possible source. This property is added as a preparation for future expansions.

Reviewer 2: ```suggestion
  Currently only `KREDITSMART` is supported as a possible event source. The property is added to be future proof for possible expansion in other areas of the platform.
```

## Trigger and frequency of push notifications

> A message will be publish regardless if the Vorgang or Antrag is updated. Therefore the activity in the Vorgang is responsible for the frequency of push notifications.

Reviewer 2: ```suggestion
A message will be published for arbitrary changes in a Vorgang or Antrag. Therefore the activity in the Vorgang is responsible for the frequency of push notifications.
```

> Currently we do not aggregate any messages, that means the client is responsible for processing the potential load of messages.

Reviewer 2: ```suggestion
Currently we do not aggregate or deduplicate any messages, so the client is responsible for handling the potential load of messages.
```

> KreditSmart does not save any push notifications. That means if there is no history of messages in intervals a subscriber is not connected to the topics.

Reviewer 2: ```suggestion
KreditSmart and the AWS broker don't store any push notifications. Consequently, a subscriber won't receive messages during unconnected time intervals.
```

## Client-Implementation

> There are a numver of different examples for how to implement and configure a MQTT client.

Reviewer 2: ```suggestion
There are a number of different examples of how to implement and configure a MQTT client.
```

> There are a number of different examples of how to implement and configure a MQTT client.

Reviewer: ```suggestion
There are a number of different examples of how to implement and configure an MQTT client.
```

> AWS provides a Java-SDK which is documented under [github.com/aws/aws-iot-device-sdk-java](https://github.com/aws/aws-iot-device-sdk-java#use-the-sdk) including sample code.

Reviewer 2: ```suggestion
AWS provides a Java-SDK which is documented at [github.com/aws/aws-iot-device-sdk-java](https://github.com/aws/aws-iot-device-sdk-java#use-the-sdk) including sample code.
```

## Terms of use

> The APIs are made available under the following [Terms of Use](https://docs.api.europace.de/terms/).

Reviewer 2: ```suggestion
The APIs are made available under our [Terms of Use](https://docs.api.europace.de/terms/).
```

Author: Diesen Text möchte ich auch ungern anfassen, da wir den schon anderweitig verwenden

Reviewer 2: Mh. Für mich suggeriert das "the following", dass ein Doppelpunkt oder Ähnliches kommen müsste, also noch etwas _nach_ "Terms of Use".
